### ISSUE TITLE
Core Contributors - Sprint grooming & planning

### ISSUE THREAD

**Issue by antoviaque** [openedx/open-edx-proposals#413]
A minimal version of core contributors sprints has actually been implemented based on https://github.com/openedx/open-edx-proposals/issues/229 , providing a base structure and rhythm, with the 2-weeks sprints, the async checkins in Listaflow and the retrospective discussions based on these reports, both in the contributor meetup and on the forums. This ticket is meant to track the future work iterating on this base, to improve and extend it.

Specifically, most of the structure in place currently is _reactive_, looking back at what has been done in the past two weeks, but there is little in the way of planning or being able to coordinate/join current efforts. This next step would build upon a more consistent use of tickets to describe current and future core contributor work, and allow to do backlog grooming and sprint planning sessions (either asynchronously or synchronously) - cf https://hygger.io/blog/product-backlog-grooming-vs-sprint-planning/

The ability to implement these steps depend on ticket creation becoming more central and systematic to the work of core contribution, which should be enabled by the following two upcoming steps:
* Concluding the discussion about what is considered "core contributor time" -- the [proposal](https://docs.google.com/document/d/1GdnhoKZymEMkvSOL4_rP9SDQrXm2utyE7LW8FIEh7OQ/edit#) currently uses the existence of tickets to define the scope, which should help encouraging it
* The initiative to make workflows more public - cf https://github.com/edx/open-source-process-wg/issues/83

To implement the grooming & planning of tickets, we could start with a synchronous session to be able to discuss and come up with a method together -- but we could also consider doing it asynchronously (OpenCraft does its planning async for 2 years now and it works well, though it requires more self-discipline than simply attending a meeting.)

**Comment by Kelketek** [comment]
Thanks for the ping on this, @antoviaque . I'll be watching closely.

So you all know, OpenCraft is transitioning to coordinating our sprint planning through Listaflow over the coming weeks. We'll update you all on how it's going, but we've been using a prototype version of the process on Monday.com for a while now. Essentially, we use a checklist every two weeks to prepare our sprint asynchronously.

This checklist includes items like:

1. Selecting your tasks for the next sprint
3. Verifying all of these tasks are properly refined
4. Reviewing your blocked tickets to see if any have been unstuck and pinging people on them if not.

It could have as many or as few tasks as the team needs to ensure get done each sprint. Of course, fewer tends to be better in terms of context switching and compliance. :)

**Comment by ali-hugo** [comment]
@antoviaque @Kelketek I'm interested to see how this develops. My only reservation would be adding yet another piece of admin for Core Contributors to complete each sprint. Would it be possible for the sprint grooming and planning to be on a less frequent basis than every sprint?

**Comment by antoviaque** [comment]
@sarina 

> I'd like to make sure there's a clear motivation and value statement (why are we doing this?) that people can agree upon. 

For me at least, this is about ensuring that:
* What I work on matches priorities -- without sprint planning work tend to be more reactive to passing urgencies, and important but not urgent, or more difficult tasks tend to be postponed
* I only have to sort priorities once in a while during sprint planning, and can just rely on that work to focus on the tasks in the sprint and forget about all the other ones -- instead of having to reload the whole backlog in my mind every time I switch tasks or a new task gets added to it
* My time is properly allocated -- proper sprint planning ensures that I will have time to do the work I select for a sprint so it isn't too hectic, and that I can also push back on sprint insertions confidently when they come in, because I already _know_ that I won't have time for them if they aren't part of a task in the current sprint
* Be able to easily advertise to others what I'm working on, and more easily provide ETAs for when it will be ready, when they depend on my work or we need to collaborate on something at a given time (eg. reviews)

> I don't want to get to the point that multiple hours of everyone's commitment is spent prepping for meetings, planning, etc. I think focusing on the "what is CC work" is the most important thing we need to do right now.

Yup, it makes sense to figure out "what is core contributor work" first, especially as the proposed measure will encourage the creation of tickets, which will be useful for implementing this. 

In terms of time investment, I don't think it requires hours -- personally I only spend about 30-35min planning all my tasks (not just the 5h/week of core contributor work) each sprint. Also, in my experience every minute spent in focused upfront planning tend to save multiple order of magnitudes that amount later on -- it's of course possible to go overboard with this, but I have yet to witness this happening imho. :)

@arbrandes Looks like a good setup to me :+1: It doesn't really require more than that imho -- but that already makes a huge difference to do this.

@Kelketek Perfect! In the meantime we'll move forward on the dependencies here, and we'll regroup. :+1: 

@ali-hugo 

> My only reservation would be adding yet another piece of admin for Core Contributors to complete each sprint (especially considering that most CC's are likely already involved in sprint planning for other teams).

It probably doesn't have to be a _different_ step though, it could be part of the current sprint check-in flow in listaflow, just expanding the items related to "What are you going to work on this sprint?"? I'm reluctant to make sprints longer than 2 weeks, it's already quite a long time - and it usually works best when the various sprints someone has to participate in can be synchronized, and that all the planning/retro work can be done at the same time (the core contributor one just being exposing a subset of it, not _additional_ work if you're already doing sprints).

**Comment by ali-hugo** [comment]
@antoviaque 
> It probably doesn't have to be a different step though, it could be part of the current sprint check-in flow in listaflow, just expanding the items related to "What are you going to work on this sprint?"

Ah, I see. That makes sense. Thanks for clarifying. I like this idea! :)

**Comment by sarina** [comment]
This issue hasn't been updated in two+ years, and this process has evolved. I'm going to close this out, and pending the result of the CC survey that ends 5/30, please open a new tracking issue in https://github.com/openedx/wg-coordination/issues
